### DISCUSSION TITLE
part-ii/2007-4/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#299] [Question comments]
Part II 2007 Number Theory: 4.I.1F 

https://questions.tripos.org/part-ii/2007-4/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

Legendre's formula states that if $r = \pi(\sqrt{x})$, then

$$
N_{r}(x) = \pi(x) - \pi(\sqrt{x}) + 1
$$

where $N_{r}(x)$ is the number of integers $\leq x$ coprime to the first $r$ primes. To see this, note that if an integer $\leq x$ is coprime to all of the primes up to $\sqrt{x}$, then it must either be equal to $1$, or it must be prime itself, and there are $\pi(x) - \pi(\sqrt{x})$ such primes. 

Now we have that $\lfloor \sqrt{48} \rfloor = 6$, so we have that

$$
\pi(48) = \pi(6) + N_{\pi(6)}(48) - 1
$$

and the primes $\leq 6$ are $2,3,5$, so we get

$$
\pi(48) = 3 + N_{3}(48) - 1 = 2 + N_{3}(48)
$$

To compute $N_{3}(48)$, we use inclusion-exclusion:

$$
\begin{align*}
N_{3}(48) &= 48 - \lfloor \frac{48}{2} \rfloor - \lfloor \frac{48}{3} \rfloor - \lfloor \frac{48}{5} \rfloor + \lfloor \frac{48}{6} \rfloor + \lfloor \frac{48}{10} \rfloor + \lfloor \frac{48}{15} \rfloor - \lfloor \frac{48}{30} \rfloor \\
&= 48 - 24 - 16 - 9 + 8 + 4 + 3 - 1 \\
&= 13 \\
\end{align*}
$$

Therefore 

$$
\pi(48) = 2 + 13 = 15
$$

</details>
